### ISSUE TITLE
Exception in thread "main" java.lang.UnsupportedOperationException: JsonObject 

### ISSUE THREAD

**Issue by fabian-schmitt** [Meghdut-Mandal/PacktPubScrapper#1]
Hey, first of all this software is a great Idea, but i somehow have issues using it.

When i follow the instructions and run the container i get the following error, see console output:

[...]
packtpubscrapper-scrapper-1    | Starting loading Other Books You May Enjoy 17
packtpubscrapper-scrapper-1    | Starting loading Index 18
packtpubscrapper-scrapper-1    | Exception in thread "main" java.lang.UnsupportedOperationException: JsonObject
packtpubscrapper-scrapper-1    |        at com.google.gson.JsonElement.getAsString(JsonElement.java:179)
packtpubscrapper-scrapper-1    |        at Scrapper$loadBookPageContent$1.invokeSuspend(Scrapper.kt:115)
packtpubscrapper-scrapper-1    |        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
packtpubscrapper-scrapper-1    |        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
packtpubscrapper-scrapper-1    |        at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:279)
packtpubscrapper-scrapper-1    |        at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
packtpubscrapper-scrapper-1    |        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
packtpubscrapper-scrapper-1    |        at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
packtpubscrapper-scrapper-1    |        at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
packtpubscrapper-scrapper-1    |        at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
packtpubscrapper-scrapper-1    |        at Scrapper.loadBookPageContent(Scrapper.kt:92)
packtpubscrapper-scrapper-1    |        at Scrapper.loadContent(Scrapper.kt:48)
packtpubscrapper-scrapper-1    |        at Scrapper.loadChapterSequential(Scrapper.kt:221)
packtpubscrapper-scrapper-1    |        at Scrapper.access$loadChapterSequential(Scrapper.kt:24)
packtpubscrapper-scrapper-1    |        at Scrapper$main$1$1$1.invokeSuspend(Scrapper.kt:200)
packtpubscrapper-scrapper-1    |        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
packtpubscrapper-scrapper-1    |        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
packtpubscrapper-scrapper-1    |        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
packtpubscrapper-scrapper-1    |        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
packtpubscrapper-scrapper-1    |        at java.base/java.lang.Thread.run(Unknown Source)

I am using WSL2 to execute docker-compose, but this shouldn't matter as far as i think.
Also authentification works, as you can see when it says "Start loading (...)".

Can you might help me?
I would also reward your effort with some donation.

**Comment by fabian-schmitt** [comment]
@Meghdut-Mandal ping

**Comment by Meghdut-Mandal** [comment]
Hey @fabian-schmitt  will have a look at this. It's been a while since I used it.

**Comment by Meghdut-Mandal** [comment]
Working on this fix #2

**Comment by molitoris** [comment]
Hey

A very cool tool!

I encounter a similar problem. It downloads the first two chapters. Afterwards, I receive a similar exception like @fabian-schmitt.

The connection is estabilished and the first two chapters are loaded. Then it breaks down :(

```
packtpubscrapper-scrapper-1  | [pool-1-thread-1] INFO org.mongodb.driver.connection - Opened connection [connectionId{localValue:4, serverValue:4}] to db:27017
packtpubscrapper-scrapper-1  | Done loading  ch02lvl1sec02
packtpubscrapper-scrapper-1  | [...]
packtpubscrapper-scrapper-1  | Done loading  ch03lvl1sec11
packtpubscrapper-scrapper-1  | [...]
packtpubscrapper-scrapper-1  | Cached ch03lvl1sec12
packtpubscrapper-scrapper-1  | Cached ch03lvl1sec13
packtpubscrapper-scrapper-1  | Cached ch03lvl1sec14
packtpubscrapper-scrapper-1  | Cached ch03lvl1sec15
packtpubscrapper-scrapper-1  | Cached ch03lvl1sec16
packtpubscrapper-scrapper-1  | Cached ch03lvl1sec17
packtpubscrapper-scrapper-1  | Cached ch03lvl1sec18
packtpubscrapper-scrapper-1  | Cached ch03lvl1sec19
packtpubscrapper-scrapper-1  | Exception in thread "main" java.lang.UnsupportedOperationException: JsonObject
packtpubscrapper-scrapper-1  |  at com.google.gson.JsonElement.getAsString(JsonElement.java:179)
packtpubscrapper-scrapper-1  |  at Scrapper$loadBookPageContent$1.invokeSuspend(Scrapper.kt:108)
packtpubscrapper-scrapper-1  |  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:138)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:112)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:14)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:62)
packtpubscrapper-scrapper-1  |  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:138)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:112)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:14)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:62)
packtpubscrapper-scrapper-1  |  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:138)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:112)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun.access$loop(SuspendFunctionGun.kt:14)
packtpubscrapper-scrapper-1  |  at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:62)
packtpubscrapper-scrapper-1  |  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
packtpubscrapper-scrapper-1  |  at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
packtpubscrapper-scrapper-1  |  at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
packtpubscrapper-scrapper-1  |  at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
packtpubscrapper-scrapper-1  |  at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
packtpubscrapper-scrapper-1  |  at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
packtpubscrapper-scrapper-1  |  at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
packtpubscrapper-scrapper-1  |  at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
packtpubscrapper-scrapper-1  |  at Scrapper.loadBookPageContent(Scrapper.kt:85)
packtpubscrapper-scrapper-1  |  at Scrapper.loadContent(Scrapper.kt:43)
packtpubscrapper-scrapper-1  |  at Scrapper.loadChapterSequential(Scrapper.kt:179)
packtpubscrapper-scrapper-1  |  at Scrapper.access$loadChapterSequential(Scrapper.kt:24)
packtpubscrapper-scrapper-1  |  at Scrapper$main$1$2$1.invokeSuspend(Scrapper.kt:157)
packtpubscrapper-scrapper-1  |  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
packtpubscrapper-scrapper-1  |  at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
packtpubscrapper-scrapper-1  |  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
packtpubscrapper-scrapper-1  |  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
packtpubscrapper-scrapper-1  |  at java.base/java.lang.Thread.run(Unknown Source)
```
